app/controllers/returns/ImportedPlasticPackagingController.scala

```diff
@@ -35,48 +36,51 @@ import scala.concurrent.{ExecutionContext, Future}
 class ImportedPlasticPackagingController @Inject() (
   override val messagesApi: MessagesApi,
   cacheConnector: CacheConnector,
-  navigator: Navigator,
   identify: IdentifierAction,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
-  form: ImportedPlasticPackagingFormProvider,
+  formProvider: ImportedPlasticPackagingFormProvider,
   val controllerComponents: MessagesControllerComponents,
-  view: ImportedPlasticPackagingView
+  view: ImportedPlasticPackagingView,
+  returnsNavigator: ReturnsJourneyNavigator
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(mode: Mode): Action[AnyContent] =
-    (identify andThen getData andThen requireData).async {
+    (identify andThen getData andThen requireData) {
       implicit request =>
-        val preparedForm = request.userAnswers.get(ImportedPlasticPackagingPage) match {
-          case None        => form()
-          case Some(value) => form().fill(value)
-        }
+        val preparedForm = request.userAnswers.fill(ImportedPlasticPackagingPage, formProvider())
+        
         request.userAnswers.get[TaxReturnObligation](ObligationCacheable) match {
-          case Some(obligation) => Future.successful(Ok(view(preparedForm, mode, obligation)))
-          case None             => Future.successful(Redirect(controllers.routes.IndexController.onPageLoad))
+          case Some(obligation) => Ok(view(preparedForm, mode, obligation))
+          case None => Redirect(controllers.routes.IndexController.onPageLoad)
         }
     }
 
   def onSubmit(mode: Mode): Action[AnyContent] =
     (identify andThen getData andThen requireData).async {
       implicit request =>
         val pptId: String = request.pptReference
+        val userAnswers = request.userAnswers
```

Comment:
could pptId and userAnswer be inline? They will not need to be call if we have a validation error. Not a blocker just a thought.